packages/ns-flashstart/files/ns-flashstart

```diff
@@ -0,0 +1,486 @@
+    sessions.raise_for_status()
+    for session in sessions.json():
+        data['sessions'].append({
+            'username': session['username'],
+            'ip': session['ip'],
+            'dns_code': session['dns_code'],
+            'dns': ','.join([session['dns']['primary_dns'], session['dns']['secondary_dns']]),
+        })
+    # jobs will be appended into a list, since it has a different format we won't use it for now
+    # however still needs to be fetched to reset the pending job counter
+    client.post(f'{FLASHSTART_API_ENDPOINT}/jobs/{username}')
+
+    return data
+
+
+def main():
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Nethesis Flashstart helper')
+    parser.add_argument('--log-level', default='info', choices=['debug', 'info', 'warning', 'error'],
+                        help='Set the log level (default: info)')
+    parser.add_argument('--syslog', action='store_true', help='Send logs to syslog')
+    subparsers = parser.add_subparsers(dest='command')
+
+    check_credentials_parser = subparsers.add_parser('check-credentials', help='Util to check Flashstart credentials')
+    check_credentials_parser.add_argument('--username', required=True)
+    check_credentials_parser.add_argument('--password', required=True)
+
+    subparsers.add_parser('sync', help='Sync Flashstart')
+    subparsers.add_parser('daemon', help='Run the sync command foreground')
+
+    args = parser.parse_args()
+    logger = logging.getLogger()
+    logger.setLevel(getattr(logging, args.log_level.upper()))
+    handler = logging.StreamHandler()
+    logger.addHandler(handler)
+    match args.command:
+        case 'check-credentials':
+            try:
+                check_credentials(args.username, hashlib.md5(args.password.encode()).hexdigest())
+                parser.exit(message='Credentials are valid.')
+            except HTTPError as e:
+                if e.response.status_code == 401:
+                    parser.exit(1, message='Invalid credentials.')
+                else:
+                    parser.exit(1, message=f'Error checking credentials: {e}')
+        case 'sync':
+            try:
+                sync()
+            except HTTPError as e:
+                logging.error(f'Error syncing profiles: {e}')
+                parser.exit(1)
+            except Exception as e:
+                logging.error(f'Unexpected error: {e}')
+                parser.exit(1)
+        case 'daemon':
+            # launch a process with the sync command once every 30 seconds, with a random delay of max 5 seconds
+            # shutoff event handling in case of SIGTERM, SIGINT or SIGHUP
+            def shutoff(_, _frame):
+                logging.info('Shutting down flashstart sync daemon')
+                event_handler.set()
+            event_handler = Event()
+            signal.signal(signal.SIGTERM, shutoff)
+
+            # enable auth cronjob
+            __enable()
```

Comment:
Proposal: remove the cronjob and add the authentication code here inside the daemon loop